posthog/api/insight.py

```diff
@@ -664,15 +667,18 @@ def destroy(self, request, *args, **kwargs):
 
         instance.delete()
 
-        log_activity(
-            organization_id=self.organization.id,
-            team_id=self.team_id,
-            user=request.user,
-            item_id=instance_id,
-            scope="Insight",
-            activity="deleted",
-            detail=Detail(name=(choose_insight_name(instance)), short_id=instance_short_id),
-        )
+        insight_name = choose_insight_name(instance)
+        # experiments creates insights with neither a name nor a derived name, don't log them
+        if insight_name:
```

Comment:
Might be a bit pick but as this is done 3 times in the same class, does it make sense to pull this out to a private method with one place commented rather than multiple?